lib/pacemaker/pcmk_sched_allocate.c

```diff
@@ -2393,6 +2392,173 @@ order_first_probes(pe_working_set_t * data_set)
     }
 }
 
+static void
+order_first_probe_then_restart_repromote(pe_action_t * probe,
+                                         pe_action_t * after,
+                                         pe_working_set_t * data_set)
+{
+    GListPtr gIter = NULL;
+    bool interleave = FALSE;
+    pe_resource_t *compatible_rsc = NULL;
+
+    if (probe == NULL || safe_str_neq(probe->task, RSC_STATUS)) {
+        return;
+    }
+
+    if (after == NULL) {
+        return;
+    }
+
+    if (probe->rsc == NULL) {
+        return;
+    }
+
+    /* Avoid running into any possible loop */
+    if (is_set(after->flags, pe_action_tracking)) {
+        return;
+    }
```

Comment:
Looking at the test changes, I realize we might want to skip the ordering if 'after' is a pseudo-action (i.e. it's an implied stop due to fencing). There might not be anything wrong with doing the ordering anyway, but my gut feeling is there's a chance of an ordering loop somewhere, and it's a case where the ordering doesn't actually help anything.

Off the top of my head I'm not sure whether the implied-by-fencing stuff has already been done by this point, but I suspect it has.

Reply:
Generally we need pseudo actions of clones/groups to go through the ordering chain. But probably we'd better somehow build dependencies only with non-pseudo actions of primitives no matter what, even if 'after' is a non-interleaved clone or a group...

Yes, native_stop_constraints() has been called in stage6(). It's stage7() at this point.

Reply:
Already have a prototype of the improvement in my hands. I'll finalize/polish the code and push it again after I get back from vacation the week after the next week.

Reply:
Sorry for the delay. Done and pushed now.